automation/src/test/java/com/microsoft/identity/common/test/automation/questions/AccessToken.java

```diff
@@ -33,26 +32,38 @@
 import net.serenitybdd.screenplay.Question;
 import net.serenitybdd.screenplay.questions.Text;
 
+import org.apache.http.util.TextUtils;
+
 public class AccessToken implements Question<String> {
+
+    private String clientId;
+    public AccessToken(String clientId){
+        this.clientId = clientId;
+    }
+
     @Override
     public String answeredBy(Actor actor) {
         String results = Text.of(Results.RESULT_FIELD).viewedBy(actor).asString();
         ReadCacheResult readCacheResult = ResultsMapper.GetReadCacheResultFromString(results);
         String accessToken = null;
         if (readCacheResult != null) {
-            User user = (User) actor;
-            user.setCacheResult(readCacheResult.tokenCacheItems.get(0));
             for (TokenCacheItemReadResult readResult : readCacheResult.tokenCacheItems) {
                 if (readResult.accessToken != null) {
-                    accessToken = readResult.accessToken;
-                    break;
+                    if (TextUtils.isEmpty(clientId) || (!TextUtils.isEmpty(clientId) && clientId.equals(readResult.clientId))) {
```

Comment:
line 51 & 52 could be merged into one 'if' statement

Reply:
It was intentional to have two if loops for readability as the latter if loop is already a bit complicated